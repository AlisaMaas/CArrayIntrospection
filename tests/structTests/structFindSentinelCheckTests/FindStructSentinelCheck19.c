#include "structHeader.h"
/**
 * This check tests we print info about both
 * array arguments when two array arguments are
 * given, and both have the same sentinel check
 *
 * We expect to find one non-optional sentinel check for f 
 * and 1 for f2.
 **/
void printc(char c) {}
int find(struct foo *f, struct foo *f2, int distance) {
	for (int i = 0; i < distance; i++) {
		if (f->string[i] == '\0' || f2->string[i] == '\0') {
			break;
		}
	}
	return 1;
}

int findArrayAccesses(struct foo *f, struct foo *f2, int distance) {
	for (int i = 0; i < distance; i++) {
		if (f[0].string[i] == '\0' || f2->string[i] == '\0') {
			break;
		}
	}
	return 1;
}

int findIndirectArrayAccesses(struct foo *f, struct foo *f2, int distance) {
	for (int i = 0; i < distance; i++) {
		if (f[0].foo->string[i] == '\0' || f2[0].foo->string[i] == '\0') {
			break;
		}
	}
	return 1;
}
