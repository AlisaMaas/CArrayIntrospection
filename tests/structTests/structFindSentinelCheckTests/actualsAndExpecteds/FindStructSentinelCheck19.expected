Printing analysis 'Promote Memory to Register' for function 'printc':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Promote Memory to Register' for function 'find':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Promote Memory to Register' for function 'findArrayAccesses':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Promote Memory to Register' for function 'findIndirectArrayAccesses':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Find each branch used to exit a loop when a sentinel value is found in an array':
Analyzing function: printc
	We found: 0 loops
Analyzing function: find
	We found: 1 loops
	Examining struct struct.foo[0] in loop for.cond
		There are 2 sentinel checks of this element in this loop
			We cannot bypass all sentinel checks for this element in this loop.
		Sentinel checks: 
			for.body
			lor.lhs.false
Analyzing function: findArrayAccesses
	We found: 1 loops
	Examining struct struct.foo[0] in loop for.cond
		There are 2 sentinel checks of this element in this loop
			We cannot bypass all sentinel checks for this element in this loop.
		Sentinel checks: 
			for.body
			lor.lhs.false
Analyzing function: findIndirectArrayAccesses
	We found: 1 loops
	Examining struct struct.foo[0] in loop for.cond
		There are 2 sentinel checks of this element in this loop
			We cannot bypass all sentinel checks for this element in this loop.
		Sentinel checks: 
			for.body
			lor.lhs.false
